ktor-client/ktor-client-core/common/src/io/ktor/client/features/HttpSend.kt

```diff
@@ -80,14 +80,25 @@ class HttpSend(
         }
     }
 
-    private class DefaultSender(private val maxSendCount: Int, private val client: HttpClient) : Sender {
+    private class DefaultSender(
+        private val maxSendCount: Int,
+        private val client: HttpClient
+    ) : Sender {
         private var sentCount: Int = 0
+        private var currentCall: HttpClientCall? = null
 
         override suspend fun execute(requestBuilder: HttpRequestBuilder): HttpClientCall {
-            if (sentCount >= maxSendCount) throw SendCountExceedException("Max send count $maxSendCount exceeded")
-            sentCount++
+            currentCall?.cancel()
+
+            if (sentCount >= maxSendCount) {
+                throw SendCountExceedException("Max send count $maxSendCount exceeded")
+            }
 
-            return client.sendPipeline.execute(requestBuilder, requestBuilder.body) as HttpClientCall
+            sentCount++
+            val call = client.sendPipeline.execute(requestBuilder, requestBuilder.body) as HttpClientCall
```

Comment:
Shouldn't we provide a more specific error message here instead of ClassCastException so that it will explain what the problem is?